astropy/wcs/utils.py

```diff
@@ -632,3 +632,213 @@ def pixel_to_skycoord(xp, yp, wcs, origin=0, mode='all', cls=None):
+
+    x, y = w_obj.wcs_world2pix(lon, lat, 1)
+    x, y = np.dot(w_obj.wcs.pc, (x-w_obj.wcs.crpix[0], y-w_obj.wcs.crpix[1]))
+
+    resids = np.concatenate((x-xo, y-yo))
+
+    return resids
+
+
+def fit_wcs_from_points(xy, world_coords, proj_point='center', 
+                        projection='TAN', sip_distortion=True, degree=4):
+    """ Given two matching sets of coordinates on detector and sky,
+        compute the WCS. Optionally, a SIP can be fit to account for geometric
+        distortion. Returns an `~astropy.wcs.WCS` object with the best fit
+        parameters for mapping between input pixel and sky coordinates.
+
+    Notes
+    ------
+    - Detector (pixel) coordinates are passed in as a tuple of arrays (`xy`),
+      and sky coordinates as a `~astropy.coordinates.SkyCoord` object.
+    - The fiducial point for the spherical projection can be set to 'center'
+      to use the mean position of input sky coordinates, or as an
+      `~astropy.coordinates.SkyCoord` object.
+    - All output will be in degrees.
+    - If the coordinate frame differs between `~astropy.coordinates.SkyCoord`
+      objects passed in for `world_coords` and `proj_point`, the frame for
+      `world_coords`  will override.
+
+    Parameters
+    ----------
+    yp : tuple of two `numpy.ndarray`
+        x & y pixel coordinates.
+    world_coords : `~astropy.coordinates.SkyCoord`
+        Skycoord object with world coordinates.
+    proj_point : 'center' or ~astropy.coordinates.SkyCoord`
+        Defaults to 'center', in which the geometric center of input world
+        coordinates will be used as the projection point. To specify an exact
+        point for the projection, a Skycoord object with a coordinate pair can
+        be passed in. For consistency, the units and frame of these coordinates
+        will be transformed to match 'world_coords' if they don't.
+
+    """
+
+    from scipy.optimize import least_squares
+    from .wcs import Sip
+    import copy
+
+    xp, yp = xy
+    lon, lat = world_coords.data.lon.deg, world_coords.data.lat.deg
+
+    # verify input
+    if (proj_point != 'center') and (type(proj_point) != type(world_coords)):
+        raise ValueError("proj_point must be set to 'center', or an" +
+                         "`~astropy.coordinates.SkyCoord` object with " +
+                         "a pair of points.")
+    if proj_point != 'center':
+        assert proj_point.size == 1
+
+    proj_codes = [
+        'AZP', 'SZP', 'TAN', 'STG', 'SIN', 'ARC', 'ZEA', 'AIR', 'CYP',
+        'CEA', 'CAR', 'MER', 'SFL', 'PAR', 'MOL', 'AIT', 'COP', 'COE',
+        'COD', 'COO', 'BON', 'PCO', 'TSC', 'CSC', 'QSC', 'HPX', 'XPH'
+    ]
+
+    if projection not in proj_codes:
+        raise ValueError("Must specify valid projection code from list of " +
+                         "supported types: ", ', '.join(proj_codes))
+
+    if type(sip_distortion) != bool:
+        raise ValueError("sip_distortion must be set to True or False.")
```

Comment:
I think so - the next 4 lines check if sip_distortion is True/False AND some other condition. If you enter any other type that evaluates to true (like a non zero int or something), it will fit SIP, but I think that behavior is misleading especially since `degree` is an input.